neps/nep-0452.md

```diff
@@ -0,0 +1,156 @@
+
+The initial discussion can be found [here](https://gov.near.org/t/official-linkdrop-standard/32463/1).
+
+## Rationale and Alternatives
+
+* Why is this design the best in the space of possible designs?
+
+This design allows for flexibility and extensibility of the standard while providing a set of criteria that cover the majority of current linkdrop use cases. The design was heavily inspired by current, functional NEPs such as the Fungible Token and Non-Fungible Token standards.
+
+* What other designs have been considered and what is the rationale for not choosing them?
+
+A generic data struct that all drop types needed to inherit from. This struct contained a name and some metadata in the form of stringified JSON. This made it easily extensible for any new types down the road.
+
+The rationale for not choosing this design was both simplicity and flexibility. Having one data struct required keys to be of one type only when in reality, they can be many at once. In addition, having a generic, open-ended metadata field could lead to many interpretations and different designs.
+
+We chose to use a KeyInfo struct that can be easily extensible and can cover all use-cases by having optional vectors of different data types. The proposed standard is simple, supports drops with multiple assets, and is backwards compatible with all previous linkdrops, and can be extended very easily.
+
+* What is the impact of not doing this?
+
+The impact of not doing this is creating a fragmented ecosystem of linkdrops, increasing the friction for user onboarding. Linkdrop claim pages (e.g. wallet providers) would have to implement custom integrations for every linkdrop provider platform. Inherently this would lead to a bad user experience when new users are onboarding and interacting with linkdrops in general.
+
+## Specification
+
+```ts
+/// Information about a specific public key.
+type KeyInfo = {
+   /// yoctoNEAR$ amount that will be sent to the claiming account (either new or existing)
+   // when the key is successfully used.
+   balance: string,
+
+   /// If using the NFT standard extension, a set of NFTData can be linked to the public key      
+   /// indicating that all those assets will be sent to the claiming account (either new or   
+   /// existing) when the key is successfully used.
+   nft_data: Vec<NFTData> | null,
+  
+   /// If using the FT standard extension, a set of FTData can be linked to the public key      
+   /// indicating that all those assets will be sent to the claiming account (either new or   
+   /// existing) when the key is successfully used.
+   ft_data: Vec<FTData> | null,
```

Comment:
Have you considered the option of having a single array of data, where each item is an enum (either FT/NFT/something else)? Not saying is better than this.

Reply:
+1 on this idea

Reply:
We have considered it but came to the conclusion that it would sacrifice the DevX for builders and through talking with some builders, it seems that the current spec makes the most sense. The main reason is that assets are nicely grouped into their own standardized fields and thus claim pages don't need to traverse an array and sort / group items based on what they want to render. 

There was an argument made that by having a standardized data type, you could render all items easily regardless if you support that standard or not. While this is true, the same can be done in this spec by just `JSON.stringify(Object.keys))`. The only difference is that by having a standardized data type, say `{ name, data }`, you would know the field names ahed of time. It would also be quite messy for repeated assets as seen here:
```
[DataType::NEAR { name: "NEAR", data: {balance: 500000}}, DataType::NFT { name: "NFT", data: { foo, bar }}, DataType::NFT { name: "NFT", data: { foo, bar }}, DataType::NFT { name: "NFT", data: { foo, bar }}]
```
rather than: 
```
{
 balance: 5000000,
nft_data: [{foo, bar}, {foo, bar}, {foo, bar}]
}
```

Reply:
Makes sense!